fix(transformer): arrow function transform alter `</this>`

overlookmotel opened the pull request:
Arrow function transform transforms `this` in `<this>` (JSX opening element), but was missing out transforming `</this>` (JSX closing element).

overlookmotel commented:
* **#3627** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/3627?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @overlookmotel and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

Dunqing commented:
Currently, the arrow function plugin has almost no test cases, can you add a test for this change?

overlookmotel committed:
fix(transformer): arrow function transform alter `</this>` (#3627)

Arrow function transform transforms `this` in `<this>` (JSX opening element), but was missing out transforming `</this>` (JSX closing element).

overlookmotel commented:
@Dunqing This has been merged this before I got a chance to add the test you requested.

To be honest, I don't think it's worthwhile adding tests at this stage for 2 reasons:

First, as noted in the code comments added in #3634, there are various problems with the transform at present, and some are not resolvable until TS transforms are modified to update the scope tree. So in my view, the best approach to testing is to later on solidify the implementation, and then enable the Babel conformance tests for arrow function transforms.

Secondly, if we do #3528, then we can remove the special handling of `<this>` in arrow function transform, as `this` as a JSX element name will just be a standard `ThisExpression` (https://github.com/oxc-project/oxc/issues/3528#issuecomment-2150843938). So tests that we add now for this will become redundant.

But if you disagree, that's fine - please just let me know, and I'll add one.
